File: frontend/src/store/services/pageObject.ts
import { api } from './api'

export interface PageObject {
  id: number
  name: string
  url?: string
  description?: string
  project_id: number
  status: 'active' | 'inactive' | 'deprecated'
  module?: string
  tags?: string[]
  page_config?: Record<string, any>
  created_by?: number
  created_at?: string
  updated_at?: string
}

export interface PageObjectCreate {
  name: string
  url?: string
  description?: string
  project_id: number
  status?: 'active' | 'inactive' | 'deprecated'
  module?: string
  tags?: string[]
  page_config?: Record<string, any>
}

export interface PageObjectUpdate {
  name?: string
  url?: string
  description?: string
  project_id?: number
  status?: 'active' | 'inactive' | 'deprecated'
  module?: string
  tags?: string[]
  page_config?: Record<string, any>
}

export interface PageObjectListParams {
  project_id?: number
  status?: string
  search?: string
  skip?: number
  limit?: number
}

export const pageObjectService = {
  // 获取页面对象列表
  async getPageObjects(params?: PageObjectListParams): Promise<PageObject[]> {
    const response = await api.get<PageObject[]>('/page-objects', { params })
    return response.data
  },

  // 获取页面对象详情
  async getPageObject(id: number): Promise<PageObject> {
    const response = await api.get<PageObject>(`/page-objects/${id}`)
    return response.data
  },

  // 创建页面对象
  async createPageObject(data: PageObjectCreate): Promise<PageObject> {
    const response = await api.post<PageObject>('/page-objects', data)
    return response.data
  },

  // 更新页面对象
  async updatePageObject(id: number, data: PageObjectUpdate): Promise<PageObject> {
    const response = await api.put<PageObject>(`/page-objects/${id}`, data)
    return response.data
  },

  // 删除页面对象
  async deletePageObject(id: number): Promise<void> {
    await api.delete(`/page-objects/${id}`)
  },
}

File: backend/app/schemas/page_object.py
"""
页面对象相关的 Pydantic 模型
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List
from datetime import datetime
from app.models.page_object import PageObjectStatus


class PageObjectBase(BaseModel):
    """页面对象基础模型"""
    name: str = Field(..., min_length=1, max_length=200, description="页面对象名称")
    url: Optional[str] = Field(None, max_length=500, description="页面URL或路径")
    description: Optional[str] = Field(None, description="页面描述")
    project_id: int = Field(..., description="项目ID")
    status: PageObjectStatus = Field(default=PageObjectStatus.ACTIVE, description="页面对象状态")
    module: Optional[str] = Field(None, max_length=100, description="模块")
    tags: Optional[List[str]] = Field(default=[], description="标签")


class PageObjectCreate(PageObjectBase):
    """创建页面对象模型"""
    page_config: Optional[Dict[str, Any]] = Field(default={}, description="页面配置")


class PageObjectUpdate(BaseModel):
    """更新页面对象模型"""
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    url: Optional[str] = Field(None, max_length=500)
    description: Optional[str] = None
    project_id: Optional[int] = None
    status: Optional[PageObjectStatus] = None
    module: Optional[str] = Field(None, max_length=100)
    tags: Optional[List[str]] = None
    page_config: Optional[Dict[str, Any]] = None


class PageObjectResponse(PageObjectBase):
    """页面对象响应模型"""
    id: int
    page_config: Optional[Dict[str, Any]] = {}
    created_by: Optional[int] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

